Problems building manif with examples in release mode

Hi there,

I receive compiler errors when following these steps on the latest version of the `devel` branch:
```
cmake -DCMAKE_BUILD_TYPE=Release -DBUILD_EXAMPLES=ON ..
make
```

The errors are of the form (similar errors appear for `SE2`, `SE3`, etc.):
```
In file included from /path/to/manif/repo/include/manif/impl/so3/SO3.h:4:0,
                 from /path/to/manif/repo/include/manif/impl/so3/SO3_map.h:4,
                 from /path/to/manif/repo/include/manif/impl/se3/SE3_base.h:6,
                 from /path/to/manif/repo/include/manif/SE3.h:9,
                 from /path/to/manif/repo/examples/se3_sam_selfcalib.cpp:183:
/path/to/manif/repo/include/manif/impl/so3/SO3_base.h: In static member function ‘static void manif::internal::AssignmentEvaluatorImpl<manif::SO3Base<_Derived> >::run_impl(const T&)’:
/path/to/manif/repo/include/manif/impl/so3/SO3_base.h:402:53: error: typedef ‘using Scalar = typename manif::SO3Base<_Derived>::Scalar’ locally defined but not used [-Werror=unused-local-typedefs]
     using Scalar = typename SO3Base<Derived>::Scalar;
```

--- a/README.md
+++ b/README.md
@@ -1,7 +1,7 @@
 # manif
 ## A small header-only library for Lie theory.
 
-[![travis](https://travis-ci.com/artivis/manif.svg?branch=devel)](https://travis-ci.com/artivis/manif)
+[![GHA](https://github.com/artivis/manif/workflows/build-and-test/badge.svg?branch=devel)](https://github.com/artivis/manif/workflows/build-and-test/badge.svg?branch=devel)
 [![appveyor](https://ci.appveyor.com/api/projects/status/l0q7b0shhonvejrd?svg=true)](https://ci.appveyor.com/project/artivis/manif)
 [![Documentation](https://codedocs.xyz/artivis/manif.svg)](https://codedocs.xyz/artivis/manif/)
 [![codecov](https://codecov.io/gh/artivis/manif/branch/devel/graph/badge.svg)](https://codecov.io/gh/artivis/manif)

--- a/include/manif/impl/se2/SE2_base.h
+++ b/include/manif/impl/se2/SE2_base.h
@@ -414,11 +414,12 @@ struct AssignmentEvaluatorImpl<SE2Base<Derived>>
   static void run_impl(const T& data)
   {
     using std::abs;
-    using Scalar = typename SE2Base<Derived>::Scalar;
-    MANIF_ASSERT(abs(data.template tail<2>().norm()-Scalar(1)) <
-                 Constants<Scalar>::eps_s,
-                 "SE2 assigned data not normalized !",
-                 invalid_argument);
+    MANIF_ASSERT(
+      abs(data.template tail<2>().norm()-typename SE2Base<Derived>::Scalar(1)) <
+      Constants<typename SE2Base<Derived>::Scalar>::eps_s,
+      "SE2 assigned data not normalized !",
+      invalid_argument
+    );
   }
 };
 

--- a/include/manif/impl/se3/SE3_base.h
+++ b/include/manif/impl/se3/SE3_base.h
@@ -469,11 +469,12 @@ struct AssignmentEvaluatorImpl<SE3Base<Derived>>
   static void run_impl(const T& data)
   {
     using std::abs;
-    using Scalar = typename SE3Base<Derived>::Scalar;
-    MANIF_ASSERT(abs(data.template tail<4>().norm()-Scalar(1)) <
-                 Constants<Scalar>::eps_s,
-                 "SE3 assigned data not normalized !",
-                 manif::invalid_argument);
+    MANIF_ASSERT(
+      abs(data.template tail<4>().norm()-typename SE3Base<Derived>::Scalar(1)) <
+      Constants<typename SE3Base<Derived>::Scalar>::eps_s,
+      "SE3 assigned data not normalized !",
+      manif::invalid_argument
+    );
   }
 };
 

--- a/include/manif/impl/se_2_3/SE_2_3_base.h
+++ b/include/manif/impl/se_2_3/SE_2_3_base.h
@@ -455,11 +455,12 @@ struct AssignmentEvaluatorImpl<SE_2_3Base<Derived>>
   static void run_impl(const T& data)
   {
     using std::abs;
-    using Scalar = typename SE_2_3Base<Derived>::Scalar;
-    MANIF_ASSERT(abs(data.template segment<4>(3).norm()-Scalar(1)) <
-                 Constants<Scalar>::eps_s,
-                 "SE_2_3 assigned data not normalized !",
-                 manif::invalid_argument);
+    MANIF_ASSERT(
+      abs(data.template segment<4>(3).norm()-typename SE_2_3Base<Derived>::Scalar(1)) <
+      Constants<typename SE_2_3Base<Derived>::Scalar>::eps_s,
+      "SE_2_3 assigned data not normalized !",
+      manif::invalid_argument
+    );
   }
 };
 

--- a/include/manif/impl/so2/SO2_base.h
+++ b/include/manif/impl/so2/SO2_base.h
@@ -330,10 +330,12 @@ struct AssignmentEvaluatorImpl<SO2Base<Derived>>
   static void run_impl(const T& data)
   {
     using std::abs;
-    using Scalar = typename SO2Base<Derived>::Scalar;
-    MANIF_ASSERT(abs(data.norm()-Scalar(1)) < Constants<Scalar>::eps_s,
-                 "SO2 assigned data not normalized !",
-                 invalid_argument);
+    MANIF_ASSERT(
+      abs(data.norm()-typename SO2Base<Derived>::Scalar(1)) <
+      Constants<typename SO2Base<Derived>::Scalar>::eps_s,
+      "SO2 assigned data not normalized !",
+      invalid_argument
+    );
   }
 };
 

--- a/include/manif/impl/so3/SO3_base.h
+++ b/include/manif/impl/so3/SO3_base.h
@@ -399,10 +399,12 @@ struct AssignmentEvaluatorImpl<SO3Base<Derived>>
   static void run_impl(const T& data)
   {
     using std::abs;
-    using Scalar = typename SO3Base<Derived>::Scalar;
-    MANIF_ASSERT(abs(data.norm()-Scalar(1)) < Constants<Scalar>::eps_s,
-                 "SO3 assigned data not normalized !",
-                 manif::invalid_argument);
+    MANIF_ASSERT(
+      abs(data.norm()-typename SO3Base<Derived>::Scalar(1)) <
+      Constants<typename SO3Base<Derived>::Scalar>::eps_s,
+      "SO3 assigned data not normalized !",
+      manif::invalid_argument
+    );
   }
 };
 

